File: Go/convert_bst.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func dfs(root *TreeNode, sum int) int {
    if root == nil {return sum}
    cur := dfs(root.Right, sum)
    root.Val += cur
    return dfs(root.Left, root.Val)
}

func convertBST(root *TreeNode) *TreeNode {
    dfs(root, 0)
    return root
}

File: Java/ConvertBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int dfs(TreeNode root, int sum) {
        if(root == null) return sum;
        int cur = dfs(root.right, sum);
        root.val += cur;
        return dfs(root.left, root.val);
    }

    public TreeNode convertBST(TreeNode root) {
        dfs(root, 0);
        return root;
    }
}